### PR TITLE
Fix broken bumpfee method

### PR DESCRIPTION
**Pull request by rajarshimaitra**
### Description

This fixes the issue I pointed at https://github.com/bitcoindevkit/bdk/issues/427. I found way around to get the `script_pubkey` for and output without a need of `get_tx()` function. 

Added one extra argument to `bump_fee` to select the utxo whose output is to be reduced.

### Notes to the reviewers

Please check the `bump_fee` function manually. I have checked them locally and it works. But there is no unit tests for these. 

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by rajarshimaitra** [comment]
Rebased on master

File: src/lib.rs [hunk 1]
```diff
@@ -855,12 +859,18 @@ where
         } => {
             let txid = Txid::from_str(txid.as_str()).map_err(|s| Error::Generic(s.to_string()))?;
 
+            let outpoint = OutPoint::new(txid, change_vout);
+            let script_pubey = match wallet.get_utxo(outpoint) {
```
Comment (notmandatory) [inline src/lib.rs:863]: small typo:
```suggestion
            let script_pubkey = match wallet.get_utxo(outpoint) {
```

File: src/lib.rs [hunk 2]
```diff
@@ -855,12 +859,18 @@ where
...
+                _ => return Err(Error::UnknownUtxo),
+            };
+
             let mut tx_builder = wallet.build_fee_bump(txid)?;
             tx_builder.fee_rate(FeeRate::from_sat_per_vb(fee_rate));
 
             if send_all {
                 // TODO: Find a way to get the recipient scriptpubkey to allow shrinking
-                //tx_builder.allow_shrinking()
+                tx_builder.allow_shrinking(script_pubey)?;
```
Comment (notmandatory) [inline src/lib.rs:873]:
```suggestion
                tx_builder.allow_shrinking(script_pubkey)?;
```

**Review by notmandatory — CHANGES_REQUESTED** [review]
LGTM but I still need to manually test. Found a small typo that should be fixed.

**Comment by rajarshimaitra** [comment]
Thanks. Updated..

**Comment by notmandatory** [comment]
I think I have a better approach in #45, please take a look and let me know what you think.

**Comment by rajarshimaitra** [comment]
Closed in favor of #45

**Comment by rajarshimaitra** [comment]
Close in favor of #45

**Pull request closed without merging**